qupath-core/src/main/java/qupath/lib/projects/Project.java

```diff
@@ -283,13 +284,29 @@ public default Manager<PixelClassifier> getPixelClassifiers() {
 	public default <S, R extends S> Manager<R> getResources(String location, Class<S> cls, String ext) {
 		return null;
 	}
-	
-	
-//	public List<String> listPixelClassifiers();
-//	
-//	public PixelClassifier loadPixelClassifier(String name);
-//	
-//	public void savePixelClassifier(String name, String PixelClassifier);
 
-	
+	/**
+	 * Store a metadata key-value pair in this project.
+	 * If a value already exists for the provided key, it will be overridden.
+	 *
+	 * @param key the key to store
+	 * @param value the value to store
+     */
+	void putMetadataValue(String key, String value);
+
+	/**
+	 * Request a metadata value stored in this project.
+	 *
+	 * @param key the key associated with the value to retrieve
+	 * @return the value associated with the provided key, or an empty Optional if no such value exists
+	 */
+	Optional<String> getMetadataValue(String key);
```

Comment:
I'm not sure `Optional` is required - the defined behavior for a `Map` is to return `null` if there is no value for the key.

The `Optional` doesn't resolve the ambiguity as to whether there is no value, or if the stored value is `null` (and there is no method to query `contains`).